frostsnap_core/src/coordinator.rs

```diff
@@ -15,21 +18,69 @@ use schnorr_fun::{
     fun::prelude::*,
     Schnorr, Signature,
 };
-use sha2::{digest::FixedOutput, Digest, Sha256};
+use sha2::Sha256;
 
 use crate::DeviceId;
 
 pub const MIN_NONCES_BEFORE_REQUEST: usize = 5;
 
 #[derive(Debug, Clone, Default)]
 pub struct FrostCoordinator {
-    keys: BTreeMap<KeyId, CoordinatorFrostKey>,
-    key_order: Vec<KeyId>,
+    keys: BTreeMap<Appkey, CoordFrostKey>,
+    key_order: Vec<Appkey>,
     action_state: Option<CoordinatorState>,
     device_nonces: BTreeMap<DeviceId, DeviceNonces>,
     mutations: VecDeque<Mutation>,
 }
 
+#[derive(Debug, Clone, bincode::Encode, bincode::Decode, PartialEq)]
+pub struct CoordFrostKey {
+    pub appkey: Appkey,
```

Comment:
Note we have the `AppKey` unencrypted so we can derive public keys but the root key which is derived from it is encrypted.